doc/specs/#5000 - Process Model 2.0/#4472 - Windows Terminal Session Management.md

```diff
@@ -0,0 +1,528 @@
+  with the current window ID and name
+
+## Future considerations
+
+* What if the user wanted to pipe a command to a pane in an existing window?
+  ```sh
+  man ping > wt -w 0 split-pane cat
+  ```
+  Is there some way for WT to pass it's stdin/out handles to the child process
+  it's creating? This is _not_ related to the current spec at hand, just
+  something the author considered while writing the spec. This likely belongs
+  over in [#492], or in its own spec.
+  - Or I suppose, with less confusion, someone could run `wt -w 0 split-pane --
+    man ping > cat`. That's certainly more sensible, and wouldn't require any
+    extra work.
+* "Single Instance Mode" is a scenario in which there is only ever one single WT
+  window. A user might want this functionality to only ever allow a single
+  terminal window to be open on their desktop. This is especially frequently
+  requested in combination with "quake mode", as discussed in [#653]. When Single
+  Instance Mode is active, and the user runs a new `wt.exe` commandline, it will
+  always end up running in the existing window, if there is one.
+
+  An earlier version of this spec proposed a new value of `glomToLastWindow`.
+  (`glomToLastWindow` was later renamed `windowingBehavior`). The `always` value
+  would disable tab tear out<sup>[[1]](#footnote-1)</sup>. It would additionally
+  disable the `newWindow` action, and prevent `wt -w new` from opening a new
+  window.
+
+  In discussion, it was concluded that this setting didn't make sense. Why did the
+  `glomToLastWindow` setting change the behavior of tear out? Single Instance Mode
+  is most frequently requested in regards to quake mode. We're leaving the
+  implementation of true single instance mode to that spec.
+* It was suggested in review that we could auto-generate names for windows, from
+  some list of words. Prior art could be the URLS for gyfcat.com, which use
+  three random words. I believe `docker` also assigns names from a random
+  selection of `adjective`+`name`. This is an interesting idea, and something
+  that could be pursued in the future.
+* We will _need_ to provide a commandline tool to list windows and their IDs &
+  names. We're thinking a list of windows, their IDs, names, PIDs, and the title
+  of the window.
+
+  Currently we're stuck with `wt.exe` which is a GUI application, and cannot
+  print to the console. Our need is now fairly high for the ability to print
+  info to the console. To remedy this, we'll need to ship another helper exe as
+  a commandline tool for working with the terminal. The design for this is left
+  for the future.
+
+## Footnotes
+
+<a name="footnote-1"><a>[1]: While tear-out is a separate track of work from
+session management in general, this setting could be implemented along with this
+set of features, and later used to control tear out as well.
+
+<a name="footnote-2"><a>[2]: Since we're reserving the keyword `new` to mean "a

```

Comment:
I still think it's not a terrible idea to say "we reserve all titles beginning with _" (like a web-browser, even though you violently hated that when Dustin mentioned it being like an href target) just so we can add more things in the future, if necessary, without stomping on existing ones. An escape hatch, if you will.

Reply:
Okay, I'll accept both `new` and "anything starting with `_`". `new` is nice and reasonable, `_` is future-proofing ourselves